packages/dashboard-core-plugins/src/linker/Linker.tsx

```diff
@@ -651,7 +661,8 @@ export class Linker extends Component<LinkerProps, LinkerState> {
 
   isColumnSelectionValid(
     panel: PanelComponent,
-    tableColumn?: LinkColumn
+    tableColumn?: LinkColumn,
+    model?: IrisGridModel
```

Comment:
This method is used by `ChartPanel` as well, which is going to have a different type of model... you should be checking within `IrisGridPanel` if the selection is valid before even passing it to this linker method, so you shouldn't need the `model` here. Put this check in `IrisGridPanel.isColumnSelectionValid`

Reply:
Right. This does unfortunately make it harder to update the link in progress type to `invalid` since I would return before calling `Linker.isColumnSelectionValid`. What do you think is the best way to approach this? Would we have to add another event, or perhaps an optional parameter to `columnSelectionValidator()` indicating if the column is already invalid?

Reply:
Let's double check with @dsmmcken and see my last comment about using `TableUtils.getNormalizedType` - I think that would make more sense than checking `model.isFilterable`.
So for array type columns (or anything of type `UNKNOWN` returns from `getNormalizedType`), we don't want to allow starting or ending a link there, because we wouldn't know how to apply that filter. So in that case we should change the cursor to `invalid` and not allow it as an option. In which case `model` is left entirely out of this, and we're just looking at the column type.

Reply:
Sounds reasonable.